modules/pam_env/pam_env.c

```diff
@@ -568,14 +575,107 @@ _pam_get_item_byname(pam_handle_t *pamh, const char *name)
   return itemval;
 }
 
+static void
+_strbuf_init(struct string_buffer *buffer)
+{
+  buffer->len = 0;
+  buffer->max = 0;
+  buffer->str = NULL;
+}
+
+static void
+_strbuf_free(struct string_buffer *buffer)
+{
+  pam_overwrite_n(buffer->str, buffer->len);
+  _pam_drop(buffer->str);
+  buffer->len = 0;
+  buffer->max = 0;
+}
+
+/*
+ * Allocates given amount of bytes in buffer for addition.
+ * Internally adding an extra byte for NUL character.
+ *
+ * Returns 0 on success, 1 on error.
+ */
+static int
```

Comment:
With C99 available, a more semantically obvious way would be to have these functions use `bool` (`stdbool.h`) for their return value.

Reply:
Would be the first use case of `bool` in linux-pam. Considering the matter of the PR, this is not the right time to add it.

Reply:
FWiW, the first use of `bool` was introduced in #94.